tests/java/TransactionProducerCli.java

```diff
@@ -216,6 +216,7 @@ public static void main (String[] args) throws Exception {
             case "6":
                 makeTestMessages(producer3, topic, 0, 0x10, 1, TransactionType.None, FlushType.Yes);
                 makeTestMessages(producer1, topic, 0, 0x20, 3, TransactionType.BeginOpen, FlushType.Yes);
+                System.exit(0);
```

Comment:
for versions > 2.3.0, `producer.close` apparently auto-aborts the transactions, whereas in earlier versions this is not the case.

Reply:
Maybe add a comment saying we dont want the transactions to be aborted